### PR TITLE
bugfix: cloudcore will exist when devicecontroller.kube.master not set

### PR DESCRIPTION
**Pull request by kadisi**
Signed-off-by: zhangjie <iamkadisi@163.com>

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind test
> /kind failing-test
> /kind feature

/kind bug

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
bugfix: cloudcore will exist when devicecontroller.kube.master not set
```

### CODE REVIEW

File: cloud/pkg/devicecontroller/config/config.go [hunk 1]
```diff
@@ -48,12 +48,17 @@ func InitConfigure() {
...
 		if err != nil {
-			errs = append(errs, fmt.Errorf("get devicecontroller.kube.master configuration key error %v", err))
+			km = ""
+			klog.Infof("can not get devicecontroller.kube.master key, use default value %v", km)
 		}
 		kc, err := config.CONFIG.GetValue("devicecontroller.kube.kubeconfig").ToString()
 		if err != nil {
 			errs = append(errs, fmt.Errorf("get devicecontroller.kube.kubeconfig configuration key error %v", err))
 		}
+		if km == "" && kc == "" {
```
Comment (fisherxu) [inline cloud/pkg/devicecontroller/config/config.go]: if run cloudcore in container, both km and kc can be empty

Comment (kadisi) [reply cloud/pkg/devicecontroller/config/config.go]: ok， i only add warnning log

File: cloud/pkg/devicecontroller/config/config.go [hunk 2]
```diff
@@ -48,12 +48,14 @@ func InitConfigure() {
 
 		km, err := config.CONFIG.GetValue("devicecontroller.kube.master").ToString()
 		if err != nil {
-			errs = append(errs, fmt.Errorf("get devicecontroller.kube.master configuration key error %v", err))
+			km = ""
+			klog.Infof("can not get devicecontroller.kube.master key, use default value %v", km)
 		}
 		kc, err := config.CONFIG.GetValue("devicecontroller.kube.kubeconfig").ToString()
 		if err != nil {
 			errs = append(errs, fmt.Errorf("get devicecontroller.kube.kubeconfig configuration key error %v", err))
```
Comment (fisherxu) [inline cloud/pkg/devicecontroller/config/config.go]: For all the config fields, It should not exit when getting value failed. We can set default value.

Comment (kadisi) [reply cloud/pkg/devicecontroller/config/config.go]: done

**Comment by subpathdev** [comment]
I think the unit test problem is given here -> the changes are looking good to me

**Comment by kadisi** [comment]
@subpathdev yes, i am waitting for pr #1301

File: cloud/pkg/edgecontroller/config/config.go [hunk 3]
```diff
@@ -250,10 +249,6 @@ func InitConfigure() {
 			kc = ""
 			klog.Infof("can not get key controller.kube.kubeconfig, use default value %v", kc)
 		}
-		if km == "" && kc == "" {
-			errs = append(errs, fmt.Errorf("controller.kube.kubeconfig and controller.kube.master are not both setd"))
-		}
-
```
Comment (chendave) [inline cloud/pkg/edgecontroller/config/config.go]: where is the new logic to check one of the [kubeconfig, kube.master] must be set?

Comment (kadisi) [reply cloud/pkg/edgecontroller/config/config.go]: if run cloudcore in container, both km and kc can be empty.

File: cloud/pkg/devicecontroller/config/config.go [hunk 4]
```diff
@@ -53,20 +51,24 @@ func InitConfigure() {
 		}
 		kc, err := config.CONFIG.GetValue("devicecontroller.kube.kubeconfig").ToString()
 		if err != nil {
-			errs = append(errs, fmt.Errorf("get devicecontroller.kube.kubeconfig configuration key error %v", err))
+			kc = ""
```
[the remark this answers is not in the document]
Comment (kadisi) [reply cloud/pkg/devicecontroller/config/config.go]: if not set kubeconfig, We need to print the info log and explain the default values

**Comment by kadisi** [comment]
#1393 has been merged.  this pr is uselss.
i close it.

**Pull request closed without merging**